parquet-hadoop/src/main/java/org/apache/parquet/hadoop/codec/SnappyDecompressor.java

```diff
@@ -147,4 +151,28 @@ public boolean needsDictionary() {
   public void setDictionary(byte[] b, int off, int len) {
     // No-op		
   }
-}
+
+  public static Object getSnappyDirectDecompressor() {
+    return Proxy.newProxyInstance(
+        SnappyDecompressor.class.getClassLoader(),
+        new Class[]{DirectCodecPool.DIRECT_DECOMPRESSION_CODEC_CLASS},
+        new InvocationHandler() {
+          @Override
+          public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+            if (method == DirectCodecPool.DECOMPRESS_METHOD) {
+              ByteBuffer src = (ByteBuffer) args[0];
+              ByteBuffer dst = (ByteBuffer) args[1];
+              if(!dst.hasRemaining()){
+                return null;
+              }
+              dst.clear();
+              int size = Snappy.uncompress(src, dst);
+              dst.limit(size);
+            }
+            throw new UnsupportedOperationException(method.toString());
+          }
+        }
+    );
+  }
```

Comment:
Do we need to implement DIRECT_DECOMPRESSION_CODEC_CLASS ?
It seems that we can use a specific implementation of CodecFactory.BytesDecompressor instead